lib/fibonacci.rb

```diff
@@ -1,8 +1,48 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
+# Time Complexity - O(1)
```

Comment:
O(1)???  So it won't take longer if n gets bigger?  hmmm...

Your solution here is O(n) both in time and space (call stack).  Otherwise well done.